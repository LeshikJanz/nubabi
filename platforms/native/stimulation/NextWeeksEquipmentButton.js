// @flow
import type { LayoutProps } from 'core/types/types';
import React from 'react';
import { Image } from 'react-native';
import { Box, Card, Text } from '../components';

const background = require('core/images/next-week-equipment.png');

type Props = {
  onPress: () => void,
};

export const NextWeeksEquipmentButton = ({ onPress }: Props) => {
  return (
    <Card padding={0} onPress={onPress}>
      <Box flex={1} alignItems="center" justifyContent="center">
        <Image
          source={background}
          style={{ flex: 1, width: 48, height: 46, marginTop: 16 }}
          resizeMode="contain"
        />
        <Box contentSpacing>
          <Text size={6} lineHeight={24} align="center">
            Equipment For
          </Text>
          <Text size={6} lineHeight={24} align="center">
            Next Week
          </Text>
        </Box>
      </Box>
    </Card>
  );
};

export default NextWeeksEquipmentButton;
