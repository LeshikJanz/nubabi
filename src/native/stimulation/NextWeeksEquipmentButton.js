// @flow
import type { LayoutProps } from '../../common/types/types';
import React from 'react';
import {
  View,
  Text,
  StyleSheet,
  Image,
  TouchableHighlight,
} from 'react-native';
import { Overlay, withLayout } from '../components';

const background = require('../../common/images/nextWeeksEquipmentButton.jpg');

type Props = {
  onPress: () => void,
  layout: LayoutProps,
};

export const NextWeeksEquipmentButton = ({ onPress, layout }: Props) => {
  const dimensions = { width: layout.parentWidth, height: layout.parentHeight };

  return (
    <TouchableHighlight
      style={styles.container}
      underlayColor="rgba(0,0,0,0)"
      onPress={onPress}
    >
      <Image
        source={background}
        style={[{ flex: 1 }, dimensions]}
        resizeMode="cover"
      >
        <Overlay>
          <Text style={styles.title}>Equipment For</Text>
          <Text style={styles.title}>Next Week</Text>
        </Overlay>
      </Image>
    </TouchableHighlight>
  );
};

const HEIGHT = 130;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    borderRadius: 4,
    height: HEIGHT,
    backgroundColor: 'red',
    shadowColor: '#000',
    shadowOpacity: 1,
    shadowRadius: 1,
    shadowOffset: {
      height: 5,
      width: 0,
    },
    overflow: 'hidden',
    alignItems: 'center',
    justifyContent: 'center',
    marginRight: 5,
  },
  overlay: {
    alignItems: 'center',
    justifyContent: 'center',
    flex: 1,
    backgroundColor: '#748294',
    opacity: 0.4,
    position: 'absolute',
    top: 0,
    marginTop: 65,
    height: HEIGHT,
    width: 200,
  },
  title: {
    fontSize: 16,
    justifyContent: 'center',
    alignItems: 'center',
    color: '#fff',
    fontWeight: '500',
    backgroundColor: 'transparent',
  },
  textContainer: {
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: 'transparent',
    flex: 1,
  },
});

export default withLayout(NextWeeksEquipmentButton);
