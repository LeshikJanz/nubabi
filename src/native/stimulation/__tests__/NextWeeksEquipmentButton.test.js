import 'react-native';
import React from 'react';
import NextWeeksEquipmentButton from '../NextWeeksEquipmentButton';
import renderer from 'react-test-renderer';

test('it renders correctly', () => {
  const tree = renderer.create(<NextWeeksEquipmentButton />).toJSON();

  expect(tree).toMatchSnapshot();
});
