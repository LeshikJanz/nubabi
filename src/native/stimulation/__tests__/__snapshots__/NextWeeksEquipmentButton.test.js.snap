// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
style = Object {
    "marginBottom": 16,
    "paddingBottom": 10,
    "paddingLeft": 10,
    "paddingRight": 10,
    "paddingTop": 10,
    "flex": 1,
    "backgroundColor": "#FFFFFF",
    "borderColor": "#9EABBC",
    "borderBottomWidth": 0,
    "borderLeftWidth": 0,
    "borderRightWidth": 0,
    "borderTopWidth": 0,
    "alignItems": "center",
    "borderBottomLeftRadius": 4,
    "borderBottomRightRadius": 4,
    "borderTopLeftRadius": 4,
    "borderTopRightRadius": 4,
    "justifyContent": "center",
    "shadowColor": "#000",
    "shadowOpacity": 0.15,
    "shadowRadius": 3,
    "shadowOffset": {
        "height": 1,
        "width": 0
    },
    "fontSize": 19.147659412894374,
    "lineHeight": 24,
    "color": "#454D56",
    "fontFamily": "SF Pro Display",
    "textAlign": "center"
}

<View
  onLayout={[Function]}
  onPress={undefined}
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": "#FFFFFF",
      "borderBottomLeftRadius": 4,
      "borderBottomRightRadius": 4,
      "borderBottomWidth": 0,
      "borderColor": "#9EABBC",
      "borderLeftWidth": 0,
      "borderRightWidth": 0,
      "borderTopLeftRadius": 4,
      "borderTopRightRadius": 4,
      "borderTopWidth": 0,
      "flex": 1,
      "justifyContent": "center",
      "marginBottom": 16,
      "paddingBottom": 0,
      "paddingLeft": 0,
      "paddingRight": 0,
      "paddingTop": 0,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.15,
      "shadowRadius": 3,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomWidth": 0,
        "borderColor": "#9EABBC",
        "borderLeftWidth": 0,
        "borderRightWidth": 0,
        "borderTopWidth": 0,
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Image
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../src/common/images/next-week-equipment.png",
        }
      }
      style={
        Object {
          "flex": 1,
          "height": 46,
          "marginTop": 16,
          "width": 48,
        }
      }
    />
    <View
      contentSpacing={true}
      style={
        Object {
          "borderBottomWidth": 0,
          "borderColor": "#9EABBC",
          "borderLeftWidth": 0,
          "borderRightWidth": 0,
          "borderTopWidth": 0,
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "borderBottomWidth": 0,
            "borderColor": "#9EABBC",
            "borderLeftWidth": 0,
            "borderRightWidth": 0,
            "borderTopWidth": 0,
            "color": "#454D56",
            "fontFamily": "SF Pro Display",
            "fontSize": 19.147659412894374,
            "lineHeight": 24,
            "textAlign": "center",
          }
        }
      >
        Equipment For
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "borderBottomWidth": 0,
            "borderColor": "#9EABBC",
            "borderLeftWidth": 0,
            "borderRightWidth": 0,
            "borderTopWidth": 0,
            "color": "#454D56",
            "fontFamily": "SF Pro Display",
            "fontSize": 19.147659412894374,
            "lineHeight": 24,
            "textAlign": "center",
          }
        }
      >
        Next Week
      </Text>
    </View>
  </View>
</View>
`;
